iotilesensorgraph/iotile/sg/graph.py

```diff
@@ -317,6 +317,21 @@ def get_tick(self, name):
         except ArgumentError:
             return 0
 
+    def add_crc(self):
+        """Check metadata to see if the sensorgraph's CRC should be added."""
+
+        if 'set_crc' not in self.metadata_database:
+            return
+
+        slot = SlotIdentifier.FromString('controller')
+        config_id = self.metadata_database['set_crc']
+        config_type, polynomial = self.get_config(slot, config_id)
+        calculated_crc = self.calculate_sensorgraph_crc(polynomial=polynomial)
```

Comment:
I would change calculate_sensorgraph_crc to a more generic call that takes an identifier defining the algorithm to use like "CRC32_104C11DB7", "SHA256" or "CRC16". Internally the generic function would use the appropriate algorithm as needed.

Reply:
Sure. I was thinking of changing it to something like `def calculate_hash(self, algorithm)`. And I guess we'll have to build some sort of known list of algorithms so that this function can match and use the appropriate one.

So its function would do something like
```
def calculate_hash(self, algorithm):
    hash_algorithm = KNOWN_ALGORITHMS[algorithm]
    hash_algorithm.calculate(binary_representation)
```

Maybe have a class like
```
class CRC:
    def __init__(self, polynomial, crc_size):
        self.polynomial = polynomial
        self.crc_size = crc_size
    def calculate(self, binary_representation):
        # actual algorithm
```

Reply:
Yea that's what I'm thinking... You could make a class that provides all the different algorithms however I will leave that to you if that's what you want to do. 

Along with the known list of algorithms, there would be an integer identifier that gets set in the configuration variable. My guess is that for the first implementation, we would have a single algorithm (this CRC32 that you have coded) and it's representation can be 0x01.